File: 2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.java
class Solution {
    public int minimumCost(int[] cost) {
        int n = cost.length;
        Arrays.sort(cost);

        int i = n-1;
        int min_amount = 0;
        while(i >= 0){
            if(i >= 0) min_amount += cost[i--];
            if(i >= 0) min_amount += cost[i--];
            if(i >= 0) i--;
        }
        return min_amount;
    }
}

File: 2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.cpp
class Solution {
public:
    int minimumCost(vector<int>& cost) {
         int n=cost.size();
        sort(cost.begin(),cost.end(),greater<int>());
        int totlcost=0;
        for(int i=0;i<n;i++){
            if((i+1)%3!=0){
              totlcost+=cost[i];  
            }
        }

       return totlcost;
        
    }
};